src/modules/0.0.6/aws_ec2_metadata/entity/instance_metadata.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  OneToOne,
+  PrimaryColumn,
+} from 'typeorm';
+
+import { cloudId, } from '../../../../services/cloud-id'
+import { Instance, } from '../../aws_ec2/entity';
+
+export enum Architecture {
+  ARM64 = "arm64",
+  I386 = "i386",
+  X86_64 = "x86_64",
+  X86_64_MAC = "x86_64_mac",
+}
+
+@Entity()
+export class InstanceMetadata {
+  // same id as the `instance` table
+  @PrimaryColumn()
+  @OneToOne(() => Instance, {
+    // deleting a row from the `instance` table deletes the corresponding row here
+    onDelete: 'CASCADE',
+  })
+  @JoinColumn({
+    name: 'id'
+  })
+  id: number;
+
+  @Column()
+  @cloudId
+  instanceId: string;
```

Comment:
hmmm but why aren't you joining by the `instanceId`? It is only possible to have values on this table if an instance already exists.

Reply:
discussed offline. `instanceId` can be nullable in the `instance` table which violates the fk constraint